frontend/src/components/pages/tasks/TaskModal.tsx

```diff
@@ -185,44 +109,199 @@ const TaskModal = ({ isOpen, setIsOpen }: Props): React.ReactElement => {
       onDelete={handleDelete}
     >
       <Flex flexDir="column" gap="20px">
+      <FormControl>
+          <FormLabel mb="5px" color="gray.main" fontWeight="700">
+            Task Type
+          </FormLabel>
+
+          <Select
+            variant="primary"
+            value={taskType}
+            onChange={(e) => setTaskType(e.target.value)}
+            border="solid"
+            borderWidth="2px"
+            borderColor="gray.300"
+            height="34px"
+          >
+            <option value="optional">Optional</option>
+            <option value="required">Required</option>
+          </Select>
+        </FormControl>
+
         <FormField
           label="Task Name"
           value={title}
           onChange={(e) => setTitle(e.target.value)}
           submitPressed={submitPressed}
-          required
         />
-
-        <FormControl isRequired mt={0} pt={0}>
+       <FormControl>
           <FormLabel mb="5px" color="gray.main" fontWeight="700">
-            Location
+            Recurrence
           </FormLabel>
 
           <Select
             variant="primary"
-            value={location}
-            onChange={(e) => setLocation(e.target.value)}
+            value={recurrence}
+            onChange={(e) => setRecurrence(e.target.value)}
             border="solid"
             borderWidth="2px"
             borderColor="gray.300"
             height="34px"
           >
-            <option value="kitchen">Kitchen</option>
-            <option value="livingRoom">Living Room</option>
-            <option value="washroom">Washroom</option>
+            <option value="Repeats">Repeats</option>
+            <option value="Single">Single</option>
           </Select>
         </FormControl>
-        <DateInput
-          dueDate={dueDate}
-          setDueDate={setDueDate}
-          dueTime={dueTime}
-          setDueTime={setDueTime}
-          isAllDay={isAllDay}
-          setIsAllDay={setIsAllDay}
-          recurrenceFrequency={recurrenceFrequency}
-          setRecurrenceFrequency={setRecurrenceFrequency}
-          submitPressed={submitPressed}
-        />
+       
+       
+       <Flex flexDir="row">
```

Comment:
Make this conditionally render depending on the recurrence
- if true, display, otherwise if not recurring, dont display
<img width="355" alt="Screenshot 2024-10-24 at 7 55 05 PM" src="https://github.com/user-attachments/assets/1cc5be77-ada9-4dec-9925-972eb9fb7dea">